packages/scxa-faceted-search-results/cypress/component/FacetedSearchContainer.cy.js

```diff
@@ -0,0 +1,105 @@
+
+  const props = {
+    results: episodes,
+    checkboxFacetGroups: [facetGroupTitles],
+    ResultsHeaderClass: ExperimentTableHeader,
+    ResultElementClass: ExperimentTableCard
+  }
+
+  it(`when results have facets they are displayed along with the results list`, () => {
+    cy.mount(<FacetedSearchContainer {...props} />)
+    cy.get(`input#facetGroupMultiSelectDropdown`)
+  })
+
+  it(`when results have no facets only the results list is displayed`, () => {
+    cy.mount(<FacetedSearchContainer {...props}
+      results={props.results.map((result) => ({element: result.element}))}/>)
+    cy.get(`input#facetGroupMultiSelectDropdown`).should(`not.exist`)
+  })
+
+  it(`clicking to select/unselect a single facet in a group works`, () => {
+    cy.mount(<FacetedSearchContainer {...props} />)
+    cy.get(`input[type="checkbox"]`).first().click()
+    cy.get(`div.small-12.medium-8.large-9.columns>div>div>div>p`)
+      .as(`selectedTitles`)
+      .should((selectedTitles) => {
+        expect(selectedTitles.length).to.be.lessThan(props.results.length)
+      })
+    cy.get(`input[type="checkbox"]`).first().click()
+    cy.get(`div.small-12.medium-8.large-9.columns>div>div>div>p`)
+      .as(`selectedTitles`)
+      .should((selectedTitles) => {
+        expect(selectedTitles.length).to.equal(props.results.length)
+      })
+  })
+
+  it(`clicking on a second facet works`, () => {
+    cy.mount(<FacetedSearchContainer {...props} />)
+    cy.get(`input[type="checkbox"]`).first().click()
+    cy.get(`div.small-12.medium-8.large-9.columns>div>div>div>p`)
+      .then((selectedTitles) => {
+        cy.get(`input[type="checkbox"]`).eq(1).click().then(() => {
+          cy.get(`div.small-12.medium-8.large-9.columns>div>div>div>p`)
+            .then((moreSelectedTitles) => {
+              expect(moreSelectedTitles.length)
+                .to.be.greaterThan(selectedTitles.length)
+            })
+        })
+      })
+  })
+
+  it(`doesn’t display duplicated facets`, () => {
+    cy.mount(<FacetedSearchContainer {...props}/>)
+
+    // TODO: simplify this messy code
+    // TODO: code fragement from the original (broken) test case
+    // wrapper.find(CheckboxFacetGroup).forEach(
+    //   (facetGroup) =>
+    //     expect(facetGroup.props().facets)
+    // .toHaveLength(uniqueFacetsByGroup[facetGroup.props().facetGroupName].length))
+    let checkboxLabels = []
+    let uniqueness = true
+    cy.contains(facetGroupTitles).siblings().get(`label`)
+      .each((label) => {
+        const labelValue = label[0].innerHTML
+        if (!checkboxLabels.includes(labelValue)) {
+          checkboxLabels.push(labelValue)
+        } else {
+          uniqueness = false
+        }
+      }).then(() => {
+        expect(uniqueness).is.true
+      })
+
+    let dropdownTitles = []
+    cy.get(`div.css-qbdosj-Input`).first().click()
+    cy.get(`div.css-dzz542-menu`).then(() => {
+      cy.get(`div.css-1n6sfyn-MenuList`).then((selectedTitles) => {
```

Comment:
I strongly encourage to use a different set of selectors here.

This is not very robust because the CSS classes’ IDs are generated by the styling framework used by React Select. If the framework decides that these IDs should be generated in a different way, or we choose to style the component differently, the test will probably break.

Another issue is that a CSS class is being used to select an element but no assertion about its style is made, so it doesn’t seem that the intent is clear.

For example, you can change L83 to:
```javascript
cy.get(`input#facetGroupMultiSelectDropdown`).first().click()
```

Reply:
That would be my exact comment, too. I don't want to use a randomly generated selector. That could break later.

Reply:
Also, if there’s no suitable alternative, [you can use regexes in CSS selectors](https://stackoverflow.com/questions/8903313/using-regular-expression-in-css). I’ve noticed that the classes have suffixes such as `-menu` or `-MenuList`.